events/quarkus/src/main/java/org/projectnessie/events/quarkus/delivery/StandardEventDelivery.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.quarkus.delivery;
+
+import io.vertx.core.Vertx;
+import java.time.Duration;
+import org.projectnessie.events.api.Event;
+import org.projectnessie.events.quarkus.config.QuarkusEventConfig;
+import org.projectnessie.events.spi.EventSubscriber;
+
+/**
+ * A Non-blocking {@link RetriableEventDelivery} that executes delivery attempts directly on Vert.x
+ * event loop. Suitable only for subscribers that do not block the event loop.
+ */
+public class StandardEventDelivery extends RetriableEventDelivery {
+
+  private final Event event;
+  private final EventSubscriber subscriber;
+  private final QuarkusEventConfig.RetryConfig config;
+  private final Vertx vertx;
+  private RetriableEventDelivery self = this;
+
+  // For testing only: 0 = not finished, 1 = success, 2 = failure, 3 = rejected
```

Comment:
Can you add some 'StateTrackingEventDelivery` to the test sources and move the `state` field there?'
Feels a bit odd to have test-only code here.

Reply:
I'm actually going to use `Mockito.spy` for that, less code to maintain :-)